src/main/java/jp/co/yahoo/yosegi/spark/utils/PartitionColumnUtil.java

```diff
@@ -17,22 +17,24 @@
  */
 package jp.co.yahoo.yosegi.spark.utils;
 
+import org.apache.spark.sql.execution.vectorized.ConstantColumnVector;
 import org.apache.spark.sql.types.StructType;
 import org.apache.spark.sql.types.StructField;
 import org.apache.spark.sql.catalyst.InternalRow;
 import org.apache.spark.sql.execution.vectorized.OnHeapColumnVector;
 import org.apache.spark.sql.execution.vectorized.ColumnVectorUtils;
+import org.apache.spark.sql.vectorized.ColumnVector;
 
 public final class PartitionColumnUtil{
 
   public static OnHeapColumnVector[] createPartitionColumns( final StructType partitionColumns , final InternalRow partitionValues , final int rowCount ){
     StructField[] fields = partitionColumns.fields();
-    OnHeapColumnVector[] vectors = new OnHeapColumnVector[ fields.length ];
+    ColumnVector[] vectors = new OnHeapColumnVector[ fields.length ];
     for( int i = 0 ; i < vectors.length ; i++ ){
       vectors[i] = new OnHeapColumnVector( rowCount , fields[i].dataType() );
-      ColumnVectorUtils.populate( vectors[i] , partitionValues , i );
+      ColumnVectorUtils.populate( (ConstantColumnVector) vectors[i] , partitionValues , i );
```

Comment:
It looks like the ClassCastException occurs because verctor[i] is an OnHeapColumnVector.
I think vectors[i] needs to be created as a ConstantColumnVector.
That requires the return value of this method to be a ColumnVector.

Reply:
Sorry. I misunderstood the specifications of ConstantColumnVector.

Reply:
I added test cases for partition.